ratis-server-api/src/main/java/org/apache/ratis/server/RaftServerConfigKeys.java

```diff
@@ -51,6 +51,16 @@ static void setStorageDir(RaftProperties properties, List<File> storageDir) {
     setFiles(properties::setFiles, STORAGE_DIR_KEY, storageDir);
   }
 
+  String STORAGE_SPACE_RESERVED_KEY = PREFIX + ".storage.space.reserved";
```

Comment:
Let's call it ".storage.space.min" or ".storage.space.required".

Reply:
Hmmm..., but 'min' or 'required' does not match the meaning of this config?
The space configured here is reserved for final snapshot usage,
but 'min' or 'required' seems to configure some space for server to consume.
|---------------for server to consume-------------------|   ~reserved~   |

Reply:
In the change below, it compares only the free spaces but not including the occupied spaces.
```root.getFreeSpace() > spaceReserved;```
The picture actually is
|------------for server to consume---------|-- existing snapshots & log files --|reserved |
So we cannot say that the space is reserved for snapshots.  Reserved space is more complicated to implement.

It seems that the amount of the space is the minimum required for starting a server.

Reply:
@szetszwo Ah, yeah, I agree with you, this should better be minRequired space for a ratis storageDir.
So what about name it to '.storage.space.min.required' ? 
(Reserved should be for other app on the same disk if there is any.)

Reply:
Sure, let's use '.storage.free-space.min'.  Thanks.

Reply:
OK.